run_dir/static/js/element_flowcell.js

```diff
@@ -0,0 +1,1374 @@
+            return this.getValue(this.demultiplex_stats, "Unassigned_Sequences", {});
+        },
+        // Re-structuring data //
+        grouped_lane_stats_pre_demultiplex() {
+            /*
+              This function groups the lane stats by lane number.
+            */
+
+            const groupedByLane = {};
+            this.lane_stats.forEach(lane => {
+                const lane_number = lane["Lane"];
+                if (!groupedByLane[lane_number]) {
+                    groupedByLane[lane_number] = {};
+                }
+                groupedByLane[lane_number] = lane;
+            });
+
+            return groupedByLane;
+        },
+        grouped_index_assignment_post_demultiplex() {
+            /* 
+              Index assignment from demultiplex info as presented by TACA
+              This function groups the info by lane number.
+            */
+            const groupedByLane = {};
+            this.index_assignment_demultiplex.forEach(sample => {
+                const lane = sample["Lane"];
+                if (!groupedByLane[lane]) {
+                    groupedByLane[lane] = [];
+                }
+                groupedByLane[lane].push(sample);
+            });
+
+            return groupedByLane;
+        },
+        // Checks //
+        lane_ids_match() {
+            /* Lane stats from the instrument might not match the demultiplexed lane stats
+              in case the instrument run was not started with a correct manifest file.
+            */
+            const keysPreDemultiplex = Object.keys(this.grouped_lane_stats_pre_demultiplex).sort();
+            const keysPostDemultiplex = Object.keys(this.grouped_index_assignment_post_demultiplex).sort();
+
+            return keysPreDemultiplex.length === keysPostDemultiplex.length &&
+                   keysPreDemultiplex.every((key, index) => key === keysPostDemultiplex[index]);        
+        },
+        demultiplex_stats_available() {
+            if (this.index_assignment_demultiplex) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    },
+    methods: {
+        // Data getters //
+        getFlowcell() {
+            axios.get("/api/v1/element_flowcell/" + this.ngi_run_id)
+                .then(response => {
+                    this.flowcell = response.data;
+                    this.flowcell_fetched = true;
+                })
+                .catch(error => {
+                    console.log(error);
+                });
+        },
+        getValue(obj, key, defaultValue = "N/A") {
+            /* A helper function to get a value from an object, with a default value if not found */
+            if (obj === null || obj == undefined || obj === "N/A") {
+                return defaultValue;
+            }
+            return obj.hasOwnProperty(key) ? obj[key] : defaultValue;
+        },
+        // Formatting //
+        barcode(sample) {
+            var barcode_str = "";
```

Comment:
`let` could be better?
```suggestion
            let barcode_str = "";
```

Reply:
Yupp, fixed this.